src/pipeline.js

```diff
@@ -95,6 +89,8 @@ class Pipeline {
     this._posts = [];
     // functions that are executed before each step
     this._taps = [];
+    // executed on error
+    this._onError = [];
```

Comment:
I'm not happy with the change to the error handling behavior. What @tripodsan has built was that error handlers were position-specific in the pipeline, so that you could overcome an issue, e.g. 
```javascript
pipe()
.before(fetch)
.error(fetch404)
.once(cont)
.error(fail)
```

so that you could differentiate between an error occurring early in the pipeline vs. late in the pipeline. Now you just have a bunch of error handlers that get called in a row, so the `fetch404` handler will get called even when the problem occurs in the `once` function.

Reply:
Quite frankly, the error handling seems overly complicated to me, and something new rather complex to learn. Can the fetch404 error only handle errors arising in the fetch steps or does it handle errors from the steps before? This is at least what I would expect…
How does this play together with before/after steps? Are `after fetch` steps inserted before or after fetch404. Is this complex error handling a capability which we actually use? 
Quite frankly, I believe we already have two mechanisms in javascript which allow for quite sophisticated error handling; one of them is promises, the other is plain javascript. Both are mechanisms that can express very complex error handling strategies; for me mechanisms like pipe() or this pipeline are mostly useful because of their simplicity – they allow me to quicker gain an understanding of the code in front of me; with increasing complexity (like error handling, before, after) we should IMO seriously consider just writing plain javascript…at least that's a system new users to helix won't have to learn…

Reply:
I dont think that new users would want to alter the pipeline with javascript. having the possiblity to catch errors and resume is an important feature. It would be sad to loose this. we debated long about this back then...I think we concluded that this behaviour was superior.

Reply:
The `.error` handler works almost exactly like `.catch` in a `Promise` chain, at least that's how it should work.

Reply:
@koraa I added a commit that re-implements the old error handling behavior. IMO, it's more cleaner this way. please review.

Reply:
This will work! :)
Why did you remove setting the http status code though? That's useful  as a default behaviour at the very least if the error is never resolved…

Reply:
> Why did you remove setting the http status code though?

I thought, I keep it the same way as it currently is in master. so far, the pipeline doesn't know about (all) the semantics of the `context` and `action`. you could potentially use the pipeline to process something else, unrelated to http :-)

> That's useful as a default behaviour at the very least if the error is never resolved…

we ensure that there will be an error set before returning to openwhisk here:
https://github.com/adobe/helix-pipeline/blob/682d4784258eac0569697ea8af8fdde4827a04a9/src/utils/openwhisk.js#L88